File: data/data.code.lua.server/AckRoleLoginOk.lua
module("AckRoleLoginOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckRoleLoginOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uname = ""
end


function encode(self)
	local pack = packet.new()
	pack:writeString(self.uname)
	return pack:encode(Msg.P_ACK_ROLE_LOGIN_OK)
end

File: data/data.code.golang.server/ack_role_login_ok.go
package proto

import (
	"packet"
)

type AckRoleLoginOk struct {
	Uname                    string
}

func (this *AckRoleLoginOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.Uname)

	return pack.Encode(P_ACK_ROLE_LOGIN_OK)
}
